tests/integration/Subscribers/SubscribersRepositoryTest.php

```diff
@@ -0,0 +1,332 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Subscribers;
+
+use DateTimeImmutable;
+use MailPoet\Entities\CustomFieldEntity;
+use MailPoet\Entities\SegmentEntity;
+use MailPoet\Entities\SubscriberCustomFieldEntity;
+use MailPoet\Entities\SubscriberEntity;
+use MailPoet\Entities\SubscriberSegmentEntity;
+use MailPoet\Models\CustomField;
+
+class SubscribersRepositoryTest extends \MailPoetTest {
+  /** @var SubscribersRepository */
+  private $repository;
+  /** @var SubscriberSegmentRepository */
+  private $subscriberSegmentRepository;
+  /** @var SubscriberCustomFieldRepository */
+  private $subscriberCustomFieldRepository;
+
+  public function _before() {
+    parent::_before();
+    $this->cleanup();
+    $this->repository = $this->diContainer->get(SubscribersRepository::class);
+    $this->subscriberSegmentRepository = $this->diContainer->get(SubscriberSegmentRepository::class);
+    $this->subscriberCustomFieldRepository = $this->diContainer->get(SubscriberCustomFieldRepository::class);
+  }
+
+  public function testItBulkTrashSubscribers(): void {
+    $subscriberOne = $this->createSubscriber('one@trash.com');
+    $subscriberTwo = $this->createSubscriber('two@trash.com');
+    $subscriberThree = $this->createSubscriber('three@trash.com');
+
+    $subscriberOneId = $subscriberOne->getId();
+    $subscriberTwoId = $subscriberTwo->getId();
+    $subscriberThreeId = $subscriberThree->getId();
+
+    $this->repository->bulkTrash([
+      $subscriberOneId,
+      $subscriberTwoId,
+    ]);
+
+    $this->entityManager->detach($subscriberOne);
```

Comment:
I've just found out that `detach` is deprecated in the latest Doctrine version and we will upgrade soon.
In these test cases I think you can just call `$this->entitiManager->clear()` instead of detaching entities one by one.